server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/routes/MailRepositoriesRoutes.java

```diff
@@ -180,7 +180,7 @@ public void defineListMails() {
             Limit limit = ParametersExtractor.extractLimit(request);
             MailRepositoryPath path = decodedRepositoryPath(request);
             try {
-                return repositoryStoreService.listMails(path, offset, limit)
+                return repositoryStoreService.listMails(path, offset, limit).map((list) -> list.collect(Guavate.toImmutableList()))
```

Comment:
Maybe we want to change the return type of `repositoryStoreService.listMails` in order to avoid that boiler plate `.map((list) -> list.collect(Guavate.toImmutableList()))` in the routes code?